middleware/protect.js

```diff
@@ -6,7 +6,7 @@ function forceLogin(keycloak, request, response) {
   var port = headerHost[1] || '';
   var protocol = request.protocol;
 
-  var redirectUrl = protocol + '://' + host + ( port === '' ? '' : ':' + port ) + request.url + '?auth_callback=1';
+  var redirectUrl = protocol + '://' + host + ( port === '' ? '' : ':' + port ) + request.originalUrl + '?auth_callback=1';
```

Comment:
I think this should be `(request.originalUrl || request.url)` to maintain backwards compatibility with pre 4.x express.